cmd/internal/flowmode/cmd_run.go

```diff
@@ -83,6 +85,8 @@ depending on the nature of the reload error.
 		},
 	}
 
+	cmd.Flags().
+		BoolVar(&r.configAutoReload, "config.auto-reload", r.configAutoReload, "Automatically reload the config file when it changes on disk")
```

Comment:
Some config files might not be hosted on a volume where file change events would be emitted (e.g., NFS). 

WDYT about also having a `--config.auto-reload.poll-interval` flag for polling for file changes in addition to the fsnotify watcher?

Reply:
Oh that's a great point - we have some NFS and FUSE systems that definitely wouldn't work here. I have a few projects I can test this with :)

Are you thinking auto-fallback to polling with a default interval, or disabling auto-reload if the filesystem doesn't support inotify events?

Reply:
> Are you thinking auto-fallback to polling with a default interval, or disabling auto-reload if the filesystem doesn't support inotify events?

Personally, I'm thinking that it's always both polling and watching for events when auto-reload is enabled, but fsnotify would potentially catch changes to files faster if the filesystem supports events.

Something like:

```go 
for {
  select {
  case <-poll.C: // re-read file based on poll 
  case event := <-watcher.Events: 
    // ... 

  // (other cases as before)
  }
}
```

Reply:
IMO for config reloading I would be fine with just polling since its simple, should almost always work and given we are polling for a single file (or folder for future) it shouldnt be a ton of overhead.

Reply:
I'll be honest, having the instant reload there is quite nice just from developing this and testing with some test configs. It looks like the default sync period on the `discovery.file` component is 10 seconds, which from personal experience, I would probably end up just end up reloading the service, especially if a misconfig brought down the agent by mistake.

@mattdurham 100% understand where you're coming from, but I'd like to do both methods - I suspect other DevOps engineers and devs running local compose agents debugging configs would appreciate the instant reload (and honestly want to work it in just for the fun of it, I'm still relatively new to Go and have only done work in Kubebuilder and Fiber and this is a joy to work on)

Reply:
I personally understand both sides of the debate here: it's nice to have near-instantaneous feedback for a great local editing loop, but using fsnotify adds some extra complexity, and may make PRs like #3855 harder to implement. 

Given file notifications aren't available on all filesystems, and fsnotify events may be dropped (which is why Promtail doesn't use them), polling is definitely required to cast the largest net, but I'm not too opinionated about whether fsnotify should runs on top of that as a convenience. 

@mattdurham Since the fsnotify code is already here, should we keep in t for now? We can always remove it in the future without a breaking change, since `--config.auto-reload` doesn't necessarily imply how auto-reload is implemented.

Reply:
I would like to see the polling method additionally added since this will default to true but fine with having that be the backup to allow the ideal case. I want to avoid the case where someone on a system that doesn't support inotify wonders why the auto-reload isn't working.